docker/bookkeeper/entrypoint.sh

```diff
@@ -48,31 +49,136 @@ export BK_tlsTrustStoreType=JKS
-  # the bookie to avoid potential conflicts.
-  echo "format bookie data and metadata"
-  /opt/bookkeeper/bin/bookkeeper shell bookieformat -nonInteractive -force -deleteCookie
-fi
-
-echo "start bookie"
-/opt/bookkeeper/scripts/entrypoint.sh bookie
+source ${SCRIPTS_DIR}/common.sh
+
+# To create directories for multiple ledgers and journals if specified
+create_bookie_dirs() {
+  IFS=',' read -ra directories <<< $1
+  for i in "${directories[@]}"
+  do
+      mkdir -p $i
+      if [ "$(id -u)" = '0' ]; then
+          chown -R "${BK_USER}:${BK_USER}" $i
+      fi
+  done
+}
+
+wait_for_zookeeper() {
+    echo "Waiting for zookeeper"
+    until zk-shell --run-once "ls /" ${BK_zkServers}; do sleep 5; done
+    echo "Done waiting for Zookeeper"
+}
+
+
+create_zk_root() {
+    if [ "x${BK_CLUSTER_ROOT_PATH}" != "x" ]; then
+        echo "Creating the zk root dir '${BK_CLUSTER_ROOT_PATH}' at '${BK_zkServers}'"
+        zk-shell --run-once "create ${BK_CLUSTER_ROOT_PATH} '' false false true" ${BK_zkServers}
+        echo "Done creating the zk root dir"
+    fi
+}
+
+configure_bk() {
+    # We need to update the metadata endpoint and Bookie ID before attempting to delete the cookie
+    sed -i "s|.*metadataServiceUri=.*\$|metadataServiceUri=${BK_metadataServiceUri}|" /opt/bookkeeper/conf/bk_server.conf
+    if [ ! -z "$BK_useHostNameAsBookieID" ]; then
+      sed -i "s|.*useHostNameAsBookieID=.*\$|useHostNameAsBookieID=${BK_useHostNameAsBookieID}|" ${BK_HOME}/conf/bk_server.conf
+    fi
+}
+
+disable_bk_ipv6_check() {
+   # This check done in "common.sh" in the Bookeeper image fails in environments where IPV6 is not enabled.
+   # We are disabling it TEMPORARILY. This needs to be reverted back/tackled in a better way, by modifying the
+   # check instead to also make sure that the file "/proc/sys/net/ipv6/bindv6only" exists, before it executes the
+   # "/sbin/sysctl -n net.ipv6.bindv6only" command.
+   sed -i "s|/sbin/sysctl|/sbin/sysctl2|" ${BK_HOME}/bin/common.sh
+}
+
+# Init the cluster if required znodes not exist in Zookeeper.
+# Use ephemeral zk node as lock to keep initialize atomic.
+function init_cluster() {
+    zk-shell --run-once "ls ${BK_zkLedgersRootPath}/available/readonly" ${BK_zkServers}
+    if [ $? -eq 0 ]; then
+        echo "Metadata of cluster already exists, no need format"
+    else
+        echo "Sleep for a random time"
+        sleep $[ ( $RANDOM % 10 )  + 1 ]s
+        # create ephemeral zk node bkInitLock, initiator who this node, then do init; other initiators will wait.
+        lock=`zk-shell --run-once "create ${BK_CLUSTER_ROOT_PATH}/bkInitLock1 '' true false false" ${BK_zkServers}`
+        echo "output of zk-shell create bkInitLock"
+        echo $lock
+        if [ -z "$lock" ]; then
+            # bkInitLock created success, this is the successor to do znode init
+            echo "Bookkeeper znodes not exist in Zookeeper, do the init to create them."
```

Comment:
The number of typos and grammatical errors in this file's strings made me think this is copied from somewhere. And indeed, this is from `init_bookie.sh` from the BK repository.

- Can we not make use of that file?
- If you do copy-paste all this code, please check out the copyright on the original file and under what conditions you are allowed (if at all) to copy the contents of your file for your own use.

Reply:
Yes, that's a good point. I've pushed a commit `033931a`, which addresses this issue.